stepflow-rs/crates/stepflow-components-mcp/src/plugin.rs

```diff
@@ -336,19 +335,27 @@ impl Plugin for McpPlugin {
     }
 
     async fn component_info(&self, component: &Component) -> Result<ComponentInfo> {
-        let tool_name = component_url_to_tool_name(component.url_string())
-            .ok_or(PluginError::ComponentInfo)
-            .attach_printable("Invalid MCP component URL format")?;
+        let tool_name = component_url_to_tool_name(component.url_string()).ok_or_else(|| {
+            error_stack::Report::new(McpError::SchemaConversion)
+                .attach_printable("Invalid MCP component URL format")
+                .change_context(PluginError::ComponentInfo)
+        })?;
 
         let state = self.state.read().await;
         let tool = state
             .available_tools
             .iter()
             .find(|tool| tool.name == tool_name)
-            .ok_or(PluginError::ComponentInfo)
-            .attach_printable("MCP tool not found")?;
+            .ok_or_else(|| {
+                error_stack::Report::new(McpError::ToolNotFound {
+                    tool_name: tool_name.to_string(),
+                })
+                .change_context(PluginError::ComponentInfo)
```

Comment:
Here and elsewhere it is possible to go straight to ComponentInfo error, not sure on the doublue change context.

Reply:
Simplified per recommendations.